http/cors.go

```diff
@@ -0,0 +1,32 @@
+package http
+
+import (
+	"net/http"
+
+	"github.com/hashicorp/vault/logical"
+	"github.com/hashicorp/vault/vault"
+)
+
+func wrapCORSHandler(h http.Handler, core *vault.Core) http.Handler {
+	return http.HandlerFunc(func(w http.ResponseWriter, req *http.Request) {
+		corsConf := core.CORSConfig()
```

Comment:
That doesn't really change the thread here: I was worried about the CORS check being done first to get it out of the hot path, and @naunga indicated that it's being done early in the function but could be moved out of it to make it more clear/explicit. I think that conversation is still valid :-)